lib/user-handler.js

```diff
@@ -432,6 +432,59 @@ class UserHandler {
             return [false, false];
         }
 
+        let isPasswordPwned = false;
+
+        if (config.pwned && config.pwned.enabled && config.pwned.type && config.pwned.type !== 'none') {
+            try {
+                isPasswordPwned = await this.checkPwnedPasswordForUser(password);
```

Comment:
If the user has not changed their password, then there is no need to re-check every time they log in. So instead of caching the result in Redis, store the result in the user object in Mongo the first time, and reset the result when account password is changed. IMAP client can create hundreds of logins in a short period of time and there is no need to check the pwned password every time it happens. Also no point in temporary caching, if the password never changes.

Reply:
New leaks do get added from time to time, though. If you only ever check once after a password changed, it makes no sense at all to check this on login, there would be no point. The only reason to check on login is because a new leak has happened, otherwise you can just check when setting the password initially.

Reply:
well, yeah, that's also valid point 🤔 the issue is, that during last 24 hours we had 9 626 141 successful authentications (and this was a regular day), so even with cashing this would be huge number of haveibeenpwnd requests